### ISSUE TITLE
uuid tests don't build as part of 'status'

### ISSUE THREAD

**Issue by beddingfield** [boostorg/uuid#60]
When trying to run all tests in boost 1.66.0, the build description for the UUID tests fail:

```
$ cd libs/uuid/test
$ ../../../bjam

Performing configuration checks
    - 32-bit                   : no  (cached)
    - 64-bit                   : yes (cached)
    - arm                      : no  (cached)
    - mips1                    : no  (cached)
    - power                    : no  (cached)
    - sparc                    : no  (cached)
    - x86                      : yes (cached)
    - symlinks supported       : yes (cached)
error: Unable to find file or target named
error:     '/libs/uuid/test//../include/boost/uuid/uuid.hpp'
error: referred to from project at
error:     '.'

```

The following patch resolves it for me:

```
diff -Nurp boost_1_66_0.orig/libs/uuid/test/Jamfile.v2 boost_1_66_0/libs/uuid/test/Jamfile.v2
--- boost_1_66_0.orig/libs/uuid/test/Jamfile.v2 2017-12-13 15:56:49.000000000 -0800
+++ boost_1_66_0/libs/uuid/test/Jamfile.v2      2018-03-16 09:06:28.569850135 -0700
@@ -41,12 +41,12 @@ rule test_all
     all_rules += [ compile compile/decl_header.cpp :
         <define>"BOOST_UUID_TEST_HEADER=uuid.hpp"
         <define>"BOOST_UUID_NO_TYPE_TRAITS"
-        <dependency>../include/boost/uuid/uuid.hpp :
+        <dependency>../../../boost/uuid/uuid.hpp :
             compile_uuid_no_type_traits ] ;
     all_rules += [ compile compile/decl_header.cpp :
         <define>"BOOST_UUID_TEST_HEADER=uuid.hpp"
         <define>"BOOST_UUID_NO_SIMD"
-        <dependency>../include/boost/uuid/uuid.hpp :
+        <dependency>../../../boost/uuid/uuid.hpp :
             compile_uuid_no_simd ] ;
     # ECHO All rules: $(all_rules) ;

```

**Comment by mclow** [comment]
A better question is WHY you don't have a file `$BOOST/libs/uuid/include/boost/uuid/uuid.hpp`

If you look here: https://github.com/boostorg/uuid/tree/master/include/boost/uuid , you should have it.

**Comment by jeking3** [comment]
Is this a case of forgetting to run "b2 headers"?

**Comment by beddingfield** [comment]
I am indeed running from the tarball, not from git.

Running "b2 headers" from the root of the source tree does not seem to make a difference.

**Comment by mclow** [comment]
Ok, then - in that case, you probably want to use (something like) `$(BOOST_ROOT)/boost/uuid/uuid.hpp` instead of a relative path.

**Comment by beddingfield** [comment]
$(BOOST_ROOT) seems to work for me

**Comment by jeking3** [comment]
That makes sense.  Thanks.  If you want to file a pull request the CI build will qualify it automatically....

**Comment by jeking3** [comment]
@danieljames given this is a build-script issue and not a code change, is it okay to fix this in develop and move it to master for 1.67.0?

**Comment by danieljames** [comment]
Sure, I'd leave it a few days to check the testers are okay, but it looks safe.

**Comment by jeking3** [comment]
I am reopening this to track the results in the matrix.  After a few days if it looks good I will merge to master for 1.67.0.

**Comment by jeking3** [comment]
@danieljames I have cherry-picked this specific fix into uuid master for 1.67.0.
